rsconnect/api.py

```diff
@@ -1,11 +1,14 @@
 """
 Posit Connect API client and utility functions
 """
+
+from __future__ import annotations
```

Comment:
On Python 3.7 and above, adding this import allows you to do the following:

- Use `list` (which does not require an import) instead of `List` (which requires `from typing import List`), and the same for `dict` and some others.\
- Use `A | B` instead of `Union[A, B]`.